src/main/java/og_spipes/service/ScriptService.java

```diff
@@ -58,30 +60,40 @@ public List<ModuleType> getModuleTypes(String filepath){
     }
 
     public void createDependency(String scriptPath, String from, String to) throws IOException {
-        Model ontModel = ontologyHelper.createOntModel(new File(scriptPath));
-        List<Resource> resources = ontModel.listSubjects().toList().stream().filter(Objects::nonNull).filter(x -> x.getURI() != null).collect(Collectors.toList());
+        Model model = ModelFactory.createDefaultModel();
+        try (InputStream is = new FileInputStream(scriptPath)) {
+            RDFDataMgr.read(model, is, Lang.TURTLE);
+        }
+        List<Resource> resources = model.listSubjects().toList().stream().filter(Objects::nonNull).filter(x -> x.getURI() != null).collect(Collectors.toList());
         Optional<Resource> moduleFrom = resources.stream().filter(x -> x.getURI().equals(from)).findAny();
         Optional<Resource> moduleTo = resources.stream().filter(x -> x.getURI().equals(to)).findAny();
 
         if(!moduleFrom.isPresent() || !moduleTo.isPresent()){
             throw new IllegalArgumentException("FROM MODULE: " + moduleFrom + " OR TO MODULE " + moduleTo + "CANT BE NULL");
         }
 
-        ontModel.add(moduleFrom.get(), new PropertyImpl(Vocabulary.s_p_next), moduleTo.get());
+        if (model.contains(moduleTo.get(), new PropertyImpl(Vocabulary.s_p_next), moduleFrom.get())) {
+            model.remove(moduleTo.get(), new PropertyImpl(Vocabulary.s_p_next), moduleFrom.get()); // Remove the connection in the opposite direction if exists
+        }
+        model.add(moduleFrom.get(), new PropertyImpl(Vocabulary.s_p_next), moduleTo.get());
+
         try (OutputStream os = new FileOutputStream(scriptPath);){
-            JenaUtils.writeScript(os, ontModel);
+            JenaUtils.writeScript(os, model);
```

Comment:
Why we can afford to load model instead of ontModel .... are imports also handled well ?